robot:
  type: RobotArm2D
  parameters:
    base_position: [5.0, 5.0]
    link_lengths: [2.1, 2.1]

workspace:
  type: Workspace2D
  parameters:
    limits: [[0, 10.0], [0, 10.0]]

    obstacles:
      - type: Rectangle
        parameters:
          xlim: [2.3, 3.4]
          ylim: [7, 9.8]

      - type: Rectangle
        parameters:
          xlim: [0.0, 4.0]
          ylim: [7.3, 8.5]

      - type: Rectangle
        parameters:
          xlim: [7.3, 8.5]
          ylim: [7.6, 10]

      - type: Rectangle
        parameters:
          xlim: [1.3, 3]
          ylim: [2.8, 3.8]

      - type: Rectangle
        parameters:
          xlim: [6.1, 8]
          ylim: [6.8, 7.5]

      - type: Polygon
        parameters:
          p_in:
            - [6.1, 8]
            - [6.8, 7.5]
            - [8, 6]
            - [6.8, 6.3]

      # This is equivalent to a Rectangle, just an example to show how other polygons could be defined by their vertices
      - type: Polygon
        parameters:
          p_in:
            - [5, 0.9]
            - [9, 0.9]
            - [7.7, 4]
            - [5, 4]

# query -> [[X1, Y1], [X2, Y2]]
queries:
  - [[1, 5], [9, 5]]
